### PR TITLE
Stats doc

### LINKED ISSUE #439: Fix jmx metrics documentation
Metrics names have changed and few new ones have been added recently, documentation need to change accordingly

### PR DESCRIPTION
**Pull request by stagraqubole**
fixes #439 

1. Fixes BookKeeperMetrics to consider non-local requests in cache hit/miss calculations
2. Added all client side stats in metrics documentation

### CODE REVIEW

File: docs/metrics.rst [hunk 1]
```diff
@@ -97,3 +99,87 @@ Metrics relating to JVM statistics, supplied by the Dropwizard Metrics ``metrics
 | rubix.bookkeeper.jvm.threads.* | Metrics relating to thread states      |               |
 | rubix.ldts.jvm.threads.*       | (CachedThreadStatesGaugeSet)           |               |
 +--------------------------------+----------------------------------------+---------------+
+
+Client Side Metrics
+-------------------
+
+These metrics are available on the client side i.e. Presto or Spark where the jobs to read data are run.
+
+Client side Metrics is divided into two:
```
Comment (harmandeeps) [inline docs/metrics.rst:108]: nit: side -> Side?

File: docs/metrics.rst [hunk 2]
```diff
@@ -97,3 +99,87 @@ Metrics relating to JVM statistics, supplied by the Dropwizard Metrics ``metrics
...
+
+1. Basic stats: These stats are available under name `rubix:name=stats`
+2. Detailed stats: These stats are available under name `rubix:name=stats,type=detailed`
+
+If Rubix is used in embedded mode an engine specific suffix is added to these names, e.g., Presto adds `catalog=<catalog_name>` suffix.
```
Comment (harmandeeps) [inline docs/metrics.rst:113]: nit: mode an -> mode, an?

File: docs/metrics.rst [hunk 3]
```diff
@@ -97,3 +99,87 @@ Metrics relating to JVM statistics, supplied by the Dropwizard Metrics ``metrics
...
+
+Following sections covers the metrics available under both these types in detail.
```
Comment (harmandeeps) [inline docs/metrics.rst:115]: covers -> cover?

File: docs/metrics.rst [hunk 4]
```diff
@@ -97,3 +99,87 @@ Metrics relating to JVM statistics, supplied by the Dropwizard Metrics ``metrics
...
+| bks_extra_data_read_in_read_through            | Extra  data read from source due to block  |
+|                                                | alignment by BookKeeper                    |
+|                                                | when parallel warmup is disabled           |
++------------------------------------------------+--------------------------------------------+
+| bks_warmup_penalty_in_read_through             | Seconds spent in copying data into cache   |
+|                                                | by BookKeeper                              |
+|                                                | when parallel warmup is disabled           |
++------------------------------------------------+--------------------------------------------+
+
+*Data units in all metrics above is MB*
```
Comment (harmandeeps) [inline docs/metrics.rst:185]: units -> unit?

File: docs/metrics.rst [hunk 5]
```diff
@@ -97,3 +99,87 @@ Metrics relating to JVM statistics, supplied by the Dropwizard Metrics ``metrics
...
+Client side Metrics is divided into two:
+
+1. Basic stats: These stats are available under name `rubix:name=stats`
+2. Detailed stats: These stats are available under name `rubix:name=stats,type=detailed`
+
+If Rubix is used in embedded mode an engine specific suffix is added to these names, e.g., Presto adds `catalog=<catalog_name>` suffix.
+
+Following sections covers the metrics available under both these types in detail.
+
+**Basic stats**
```
Comment (harmandeeps) [inline docs/metrics.rst:117]: nit: stats -> Stats

File: docs/metrics.rst [hunk 6]
```diff
@@ -80,8 +83,7 @@ Metrics relating to cache interactions.
 |                                                |                                            |                                |
 +------------------------------------------------+--------------------------------------------+--------------------------------+
 
-JVM
----
+**JVM Metric**
```
Comment (harmandeeps) [inline docs/metrics.rst:86]: nit: Metric -> Metrics?

File: docs/metrics.rst [hunk 7]
```diff
@@ -97,3 +99,87 @@ Metrics relating to JVM statistics, supplied by the Dropwizard Metrics ``metrics
...
++------------------------------------------------+--------------------------------------------+
+| cache_hit                                      | Cache Hit ratio, between 0 and 1           |
++------------------------------------------------+--------------------------------------------+
+
+**Detailed Stats**
+
++------------------------------------------------+--------------------------------------------+
+| Metric                                         | Description                                |
++================================================+============================================+
+| mb_read_from_cache                             | Data read from Cache                       |
```
Comment (harmandeeps) [inline docs/metrics.rst:134]: Cache -> cache?

File: docs/metrics.rst [hunk 8]
```diff
@@ -97,3 +99,87 @@ Metrics relating to JVM statistics, supplied by the Dropwizard Metrics ``metrics
...
++------------------------------------------------+--------------------------------------------+
+| mb_read_from_source                            | Data read from Source                      |
++------------------------------------------------+--------------------------------------------+
+| cache_hit                                      | Cache Hit ratio, between 0 and 1           |
++------------------------------------------------+--------------------------------------------+
+| cached_rrc_data_read                           | Data read from local Cache                 |
```
Comment (harmandeeps) [inline docs/metrics.rst:140]: Cache -> cache?

File: docs/metrics.rst (hunk 1 above)
Comment (stagraqubole) [reply docs/metrics.rst:108]: change header instead of use lower case

File: docs/metrics.rst (hunk 5 above)
Comment (stagraqubole) [reply docs/metrics.rst:117]: left it lower case

**Pull request merged**